app/services/theme.js

```diff
@@ -22,9 +23,22 @@ export default Ember.Service.extend({
         return `${config.OSF.url}register?${query}`;
     }),
 
+    pathPrefix: Ember.computed('isProvider', 'domain', 'id', function() {
```

Comment:
Once again, I think `pathPrefix` is pretty generic/unclear. How about renaming this `addPreprintUrl` and making it return the full path, including `/submit/`, and pass that to the navbar?